novawalletTests/Common/Helpers/ChainFilterStrategyTests.swift

```diff
@@ -0,0 +1,365 @@
+        hasProxyFilterDeletes { updatedChains in
+            updatedChains.map { DataProviderChange<ChainModel>.update(newItem: $0) }
+        }
+        
+        hasProxyFilterDeletes { updatedChains in
+            updatedChains.map { DataProviderChange<ChainModel>.insert(newItem: $0) }
+        }
+    }
+    
+    func hasProxyFilterDeletes(changes: ([ChainModel]) -> [DataProviderChange<ChainModel>]) {
+        // given
+        let filterStrategy = ChainFilterStrategy.hasProxy
+        
+        var currentChains = [
+            ChainModelGenerator.generateChain(assets: [], addressPrefix: 0, hasProxy: true),
+            ChainModelGenerator.generateChain(assets: [], addressPrefix: 0, hasProxy: true)
+        ].reduceToDict()
+        
+        let updatedChains = currentChains.values.map {
+            ChainModelGenerator.generateChain(
+                defaultChainId: $0.chainId,
+                generatingAssets: 0,
+                addressPrefix: 0,
+                hasProxy: !$0.hasProxy
+            )
+        }
+        
+        let beforeFilterChanges = changes(updatedChains)
+        
+        // when
+        let resultChanges = filterStrategy.filter(beforeFilterChanges, using: currentChains)
+        currentChains = resultChanges.mergeToDict(currentChains)
+        
+        // then
+        XCTAssert(currentChains.isEmpty)
+    }
+    
+    func testHasProxyFilterInsert() {
+        hasProxyFilterInserts { updatedChains in
+            updatedChains.map { DataProviderChange<ChainModel>.update(newItem: $0) }
+        }
+        
+        hasProxyFilterInserts { updatedChains in
+            updatedChains.map { DataProviderChange<ChainModel>.insert(newItem: $0) }
+        }
+    }
+    
+    func hasProxyFilterInserts(changes: ([ChainModel]) -> [DataProviderChange<ChainModel>]) {
+        // given
+        let filterStrategy = ChainFilterStrategy.hasProxy
+        
+        var currentChains = [
+            ChainModelGenerator.generateChain(assets: [], addressPrefix: 0, hasProxy: false),
+            ChainModelGenerator.generateChain(assets: [], addressPrefix: 0, hasProxy: false)
+        ].reduceToDict()
+        
+        var resultChains: [ChainModel.Id: ChainModel] = [:]
+        
+        let updatedChains = currentChains.values.map {
+            ChainModelGenerator.generateChain(
+                defaultChainId: $0.chainId,
+                generatingAssets: 0,
+                addressPrefix: 0,
+                hasProxy: !$0.hasProxy
+            )
+        }
+        
+        let beforeFilterChanges = changes(updatedChains)
+        
+        // when
+        let resultChanges = filterStrategy.filter(beforeFilterChanges, using: currentChains)
+        resultChains = resultChanges.mergeToDict(resultChains)
+        
+        // then
+        XCTAssert(updatedChains.allSatisfy { resultChains[$0.chainId] != nil })
+    }
+    
+    // MARK: enabledChains
+    
+    func testEnabledChainsFilterDelete() {
```

Comment:
Didn't get why it is called ```enabled chains``` test but we actually test proxy filter. Probably it is intended to call ```enabledChainsFilterDeletes``` not proxy one

Reply:
Oh, thanks! Yes, you are right. A bit lack of attention on my side.